### PR TITLE
feat: notification settings

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Query and mutation for the new schema for managing device notification preferences
- Query and mutation for the new schema for managing general notification preferences
- New account page dedicated to user notifications preferences
- Fixed the checkbox disabled state styling
- I don't know why some `RankModal` changes are reflected here, kindly disregard them.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/lib/image.ts [hunk 1]
```diff
@@ -10,4 +10,9 @@ export const cloudinary = {
     supercharge:
       'https://daily-now-res.cloudinary.com/image/upload/v1665563545/public/supercharge.png',
   },
+  notifications: {
+    browser:
+      'https://daily-now-res.cloudinary.com/image/upload/v1670221485/public/notification.png',
+    bell: '',
```
Comment (rebelchris) [inline packages/shared/src/lib/image.ts:16]: Please sort this one out before merging :)

File: packages/webapp/__tests__/AccountNotificationsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  );
+  expect(newSubscription).toHaveAttribute('data-testValue', 'true');
+  const marketingSubscription = await screen.findByTestId('marketing-switch');
+  expect(marketingSubscription).toHaveAttribute('data-testValue', 'true');
+  const notificationEmail = await screen.findByTestId('new_activity-switch');
+  expect(notificationEmail).toHaveAttribute('data-testValue', 'true');
+  expect(mutationCalled).toBeTruthy();
+});
+
+// when you comment all tests this work
```
Comment (rebelchris) [inline packages/webapp/__tests__/AccountNotificationsPage.tsx:192]: Why is this still here?

File: packages/webapp/pages/account/notifications.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+const ALERT_PUSH_KEY = 'alert_push_key';
+
+const AccountNotificationsPage = (): ReactElement => {
+  const { user, visit } = useContext(AuthContext);
+  const [isAlertShown, setIsAlertShown] = usePersistentContext(
+    ALERT_PUSH_KEY,
+    true,
+  );
+  const { data: generalData, refetch: refetchGeneralData } = useQuery(
```
Comment (rebelchris) [inline packages/webapp/pages/account/notifications.tsx:36]: Should we maybe abstract these queries into a lib function?

File: packages/webapp/pages/account/notifications.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+          />
+          <div className="flex overflow-hidden relative flex-row p-4">
+            <p className="flex-1 break-words typo-subhead">
+              Switch on push notifications so you never miss exciting
+              discussions, upvotes, replies or mentions on daily.dev!
+            </p>
+            <img
+              className="absolute top-0 right-10"
+              src={cloudinary.notifications.browser}
+              alt="Preview"
```
Comment (rebelchris) [inline packages/webapp/pages/account/notifications.tsx:127]: Perhaps we should inline this to the other alt we used in the previous pr?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice work, couple of minor changes 🙏

File: packages/shared/src/lib/image.ts (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/lib/image.ts:16]: This is not used here. Just the browser. The bell one is found on the other PR.

File: packages/webapp/__tests__/AccountNotificationsPage.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/__tests__/AccountNotificationsPage.tsx:192]: Oh, tests issues. Lol. This is okay now 🤣

File: packages/webapp/pages/account/notifications.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/webapp/pages/account/notifications.tsx:127]: Yep. Right way to go. Let me refactor this.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Just one cleanup task left 🛳️

**Review by rebelchris — APPROVED** [review]
Sorry see it's already sorted 🥳

**Pull request merged**